src/main/java/bio/terra/cli/service/utils/SamService.java

```diff
@@ -146,6 +151,238 @@ public void getOrRegisterUser() {
     }
   }
 
+  /**
+   * Call the SAM "/api/groups/v1/{groupName}" POST endpoint to create a new group.
+   *
+   * @param groupName name of the new group
+   */
+  public void createGroup(String groupName) {
+    GroupApi groupApi = new GroupApi(apiClient);
+    try {
+      HttpUtils.callWithRetries(
+          () -> {
+            groupApi.postGroup(groupName);
+            return null;
+          },
+          SamService::isRetryable);
+    } catch (ApiException | InterruptedException ex) {
+      throw new RuntimeException("Error creating SAM group.", ex);
+    } finally {
+      closeConnectionPool();
+    }
+  }
+
+  /**
+   * Call the SAM "/api/groups/v1/{groupName}" DELETE endpoint to delete an existing group.
+   *
+   * @param groupName name of the group to delete
+   */
+  public void deleteGroup(String groupName) {
+    GroupApi groupApi = new GroupApi(apiClient);
+    try {
+      HttpUtils.callWithRetries(
+          () -> {
+            groupApi.deleteGroup(groupName);
+            return null;
+          },
+          SamService::isRetryable);
+    } catch (ApiException | InterruptedException ex) {
+      throw new RuntimeException("Error deleting SAM group.", ex);
+    } finally {
+      closeConnectionPool();
+    }
+  }
+
+  /**
+   * Call the SAM "/api/groups/v1/{groupName}" GET endpoint to get the email address of a group.
+   *
+   * @param groupName name of the group to delete
+   */
+  public String getGroupEmail(String groupName) {
+    GroupApi groupApi = new GroupApi(apiClient);
+    try {
+      return HttpUtils.callWithRetries(() -> groupApi.getGroup(groupName), SamService::isRetryable);
+    } catch (ApiException | InterruptedException ex) {
+      throw new RuntimeException("Error getting email address of SAM group.", ex);
+    } finally {
+      closeConnectionPool();
+    }
+  }
+
+  /**
+   * Possible values for the policies on a SAM group. These values are defined as an enum in SAM's
+   * API YAML
+   * (https://github.com/broadinstitute/sam/blob/61135c798873d20a308be1e440b862bf9767c243/src/main/resources/swagger/api-docs.yaml#L383)
+   * but I don't see an enum in the client library. It looks like that was a bug in Swagger codegen
+   * until v2.1.5 (https://github.com/swagger-api/swagger-codegen/pull/1740), but I'm not sure if
+   * that applies to the version that SAM is using.
+   */
```

Comment:
not urgent, but might be good for us to eventually file a few tickets for all the issues we've encountered with the Sam client library. Expectations may vary, but probably still the right thing to do :/

Reply:
I would like to talk to the CI team about some of these things -- especially the pet SA key fetch. But I just don't have the bandwidth to follow up on them right now.